play-scala-secure-session-example/app/controllers/package.scala

```diff
@@ -34,7 +34,7 @@ package object controllers {
   val form = Form(
     mapping(
       "username" -> text
-    )(UserInfo.apply)(UserInfo.unapply)
+    )(UserInfo.apply)(t => Some(t.toString()))
```

Comment:
Just curious, why doesn't `unapply` work here? Also is `Some(_.toString())` doable here?

Reply:
If I keep `unapply` like initialy I have this error:

```
[error] -- [E007] Type Mismatch Error: /Users/sachagarnier/Codes/fork/play-samples/play-scala-secure-session-example/app/controllers/package.scala:38:22
[error] 38 |    )(UserInfo.apply)(UserInfo.unapply)
[error]    |                      ^^^^^^^^^^^^^^^^
[error]    |                      Found:    controllers.package.UserInfo
[error]    |                      Required: Option[String]
```
If I try with _ in the parameters it doesn't change anything:
```
[error] -- [E007] Type Mismatch Error: /Users/sachagarnier/Codes/fork/play-samples/play-scala-secure-session-example/app/controllers/package.scala:38:38
[error] 38 |    )(UserInfo.apply)(UserInfo.unapply(_))
[error]    |                      ^^^^^^^^^^^^^^^^^^^
[error]    |                      Found:    controllers.package.UserInfo
[error]    |                      Required: Option[String]
```

If I try to match with its requirement (with an ugly way) I obtain this error:
```
[error] -- [E007] Type Mismatch Error: /Users/sachagarnier/Codes/fork/play-samples/play-scala-secure-session-example/app/controllers/package.scala:38:50
[error] 38 |    )(UserInfo.apply)(t => UserInfo.unapply(Option(t.toString)))
[error]    |                                            ^^^^^^^^^^^^^^^^^^
[error]    |                                  Found:    Option[String]
[error]    |                                  Required: controllers.package.UserInfo
```

With the second solution I have this issue: 
```
[error] -- [E081] Type Error: /Users/sachagarnier/Codes/fork/play-samples/play-scala-secure-session-example/app/controllers/package.scala:37:27
[error] 37 |    )(UserInfo.apply)(Some(_.toString()))
[error]    |                           ^
[error]    |   Missing parameter type
[error]    |
[error]    |   I could not infer the type of the parameter _$1 of expanded function:
[error]    |   _$1 => _$1.toString().
```